src/app/courses/course-item/course-item.component.spec.ts

```diff
@@ -1,25 +1,115 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { Course } from './../../shared/courses/course.interface';
+import { DebugElement, Component } from '@angular/core';
+import { ComponentFixture, TestBed, waitForAsync} from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
 
 import { CourseItemComponent } from './course-item.component';
+import { DurationPipe } from '../../duration.pipe';
+import { DatePipe } from '@angular/common';
+
+import { click } from './../../shared/helpers/spec/clickEvent';
+
+// mock the course supplied by the parent component
+const expectedCourse: Course = {
+  id: 1,
+  title: 'Course 1. This is title.',
+  dateCreation: new Date(),
+  duration: 88,
+  description: 'Learn about where you can find course descriptions, what information they include, how they work, and details about various components of a course description. Course descriptions report information about a university or college\'s classes. They\'re published both in course catalogs that outline degree requirements and in course schedules that contain descriptions for all courses offered during a particular semester.'
+};
+@Component({
+  template: `
+    <app-course-item
+      (deleteCourse)="onDelete($event)"
+      [course]="course"></app-course-item>`
+})
+class TestHostComponent {
+  course: Course = expectedCourse;
+  onDelete(id: number): void {}
+}
 
 describe('CourseItemComponent', () => {
   let component: CourseItemComponent;
   let fixture: ComponentFixture<CourseItemComponent>;
+  let courseDe: DebugElement;
+  let courseEl: HTMLElement;
 
-  beforeEach(async () => {
-    await TestBed.configureTestingModule({
-      declarations: [ CourseItemComponent ]
-    })
-    .compileComponents();
-  });
-
-  beforeEach(() => {
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [ CourseItemComponent, DurationPipe ]
+    });
     fixture = TestBed.createComponent(CourseItemComponent);
     component = fixture.componentInstance;
+
+    // find the course's DebugElement and element
+    courseDe = fixture.debugElement.query(By.css('.course-item'));
+    courseEl = courseDe.nativeElement;
+
+    // simulate the parent setting the input property with that course
+    component.course = expectedCourse;
+
+    // trigger initial data binding
     fixture.detectChanges();
-  });
+  }));
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should display course title', () => {
+    const expectedCourseTitle = expectedCourse.title;
+    const courseTitle = fixture.debugElement.query(By.css('.course-item__title'));
```

Comment:
you can create a helper function to avoid repeating yourself. e.g.
```typescript
export function getDebugElement(fixture: ComponentFixture<any>, cssSelector: string): DebugElement {
  return fixture.debugElement.query(By.css(cssSelector));
}
```

Reply:
done